pkg/aws/cloud.go

```diff
@@ -132,6 +131,20 @@ func NewCloud(cfg CloudConfig, metricsRegisterer prometheus.Registerer) (Cloud,
 	}, nil
 }
 
+func getVpcID(cfg CloudConfig, ec2Service services.EC2, metadata services.EC2Metadata, logger logr.Logger) (string, error) {
+
+	if cfg.VpcID != "" {
+		logger.V(1).Info("vpcid is specified using flag --aws-vpc-id, controller will use the value", "vpc: ", cfg.VpcID)
+		return cfg.VpcID, nil
+	}
+
+	if cfg.VpcTags != nil {
+		return inferVPCIDFromTags(ec2Service, cfg.VpcNameTagKey, cfg.VpcTags[cfg.VpcNameTagKey])
```

Comment:
@jeswinkoshyninan I don't really get this part: why does this allow to pass a map with an arbitrary number of items for `--aws-vpc-tags` if we only ever pass a single key-value pair (the one with the key specified in `aws-vpc-tag-key`) as the tag filter? Shouldn't it rather just convert the map from `--aws-vpc-tags` into a list of tag filters, one for each item?

Reply:
Related to #3889